app/src/main/java/org/jellyfin/androidtv/ui/playback/overlay/CustomPlaybackTransportControlGlueHelper.java

```diff
@@ -0,0 +1,460 @@
+    static final boolean DEBUG = false;
+
+    static final int MSG_UPDATE_PLAYBACK_STATE = 100;
+    static final int UPDATE_PLAYBACK_STATE_DELAY_MS = 2000;
+
+    private TextView endsAt = null;
+
+    PlaybackSeekDataProvider mSeekProvider;
+    boolean mSeekEnabled;
+
+    static class UpdatePlaybackStateHandler extends Handler {
+        @Override
+        @SuppressWarnings("unchecked")
+        public void handleMessage(Message msg) {
+            if (msg.what == MSG_UPDATE_PLAYBACK_STATE) {
+                CustomPlaybackTransportControlGlueHelper glue =
+                        ((WeakReference<CustomPlaybackTransportControlGlueHelper>) msg.obj).get();
+                if (glue != null) {
+                    glue.onUpdatePlaybackState();
+                }
+            }
+        }
+    }
+
+    static final Handler sHandler = new UpdatePlaybackStateHandler();
+
+    final WeakReference<CustomPlaybackTransportControlGlueHelper> mGlueWeakReference =
+            new WeakReference<>(this);
+
+    /**
+     * Constructor for the glue.
+     *
+     * @param context
+     * @param impl Implementation to underlying media player.
+     */
+    public CustomPlaybackTransportControlGlueHelper(Context context, T impl) {
+        super(context, impl);
+    }
+
+    @Override
+    public void setControlsRow(PlaybackControlsRow controlsRow) {
+        super.setControlsRow(controlsRow);
+        sHandler.removeMessages(MSG_UPDATE_PLAYBACK_STATE, mGlueWeakReference);
+        onUpdatePlaybackState();
+    }
+
+    @Override
+    protected void onCreatePrimaryActions(ArrayObjectAdapter primaryActionsAdapter) {
+        primaryActionsAdapter.add(mPlayPauseAction =
+                new PlaybackControlsRow.PlayPauseAction(getContext()));
+    }
+
+    @Override
+    protected PlaybackRowPresenter onCreateRowPresenter() {
+        final AbstractDetailsDescriptionPresenter detailsPresenter =
+                new AbstractDetailsDescriptionPresenter() {
+                    @Override
+                    protected void onBindDescription(ViewHolder
+                                                             viewHolder, Object obj) {
+                        CustomPlaybackTransportControlGlueHelper glue = (CustomPlaybackTransportControlGlueHelper) obj;
+                        viewHolder.getTitle().setText(glue.getTitle());
+                        viewHolder.getSubtitle().setText(glue.getSubtitle());
+                    }
+                };
+
+        PlaybackTransportRowPresenter rowPresenter = new PlaybackTransportRowPresenter() {
+            @Override
+            protected void onBindRowViewHolder(RowPresenter.ViewHolder vh, Object item) {
+                super.onBindRowViewHolder(vh, item);
+                vh.setOnKeyListener(CustomPlaybackTransportControlGlueHelper.this);
+                endsAt = new TextView(getContext());
+                endsAt.setText("Test");
+                LinearLayout view = (LinearLayout)vh.view;
```

Comment:
Here is the start of the custom code. I store the top row of buttons to a view, create a RelativeLayout, add the buttons to the relative layout along with the "ends at" text then add the relative layout back to the playback control view.